### PR TITLE
Implement Flatpickr

### PR DESCRIPTION
**Pull request by jinjie**
Attempt to implement Flatpickr on calendar fields.

Refer to #301

### CODE REVIEW

File: app/views/pages/report.ejs [hunk 1]
```diff
@@ -499,6 +499,7 @@
   </div>
 </section>
 
+<script src="https://cdn.jsdelivr.net/npm/flatpickr"></script>
```
Comment (adriaandotcom) [inline app/views/pages/report.ejs]: Please use https://www.srihash.org/ to include the SRI hash for external resources.

File: static/form.js [hunk 2]
```diff
@@ -9,6 +9,8 @@ function changeTestResultDisplay(show) {
   }
 }
 
+flatpickr(document.querySelectorAll('[type=date]'));
```
Comment (adriaandotcom) [inline static/form.js]: Please put it in a different script tag so the other code will not fail when this resource is not loaded. For whatever reason ;)

**Review by adriaandotcom — CHANGES_REQUESTED** [review]
Nice work! Would you mind checking these two comments?

**Comment by fossecode** [comment]
Super nice! A couple of things to think about:
- Should we format the date that is displayed to the user after it is selected? Seems like Flatpickr has support for this. We could just add the format to `/locales/xx.json`.
- @adriaanvanrossum ok to use the CDN, or should we host the files ourselves?
- Could you try to fix the input field on iOS devices? Don't know if it looks the same on Androids.
IMG_2763

**Comment by adriaandotcom** [comment]
Let's just put it on our CDN. It's privacy friendly and they don't track anything.

https://cdn.coronastatus.eu. It's backed by BunnyCDN.

**Comment by fossecode** [comment]
Also, for each new resource we add in `<head>` the time before something is displayed to user increases. Some resources are not required on all pages, and we should either: 
- Load them in the footer
- Only load them on the pages where they are required, but then we have to change `head.ejs` a bit.

We can look at that in a separate issue, not crucial for this PR.

File: app/views/pages/report.ejs (hunk 1 above)
Comment (jinjie) [reply app/views/pages/report.ejs]: Do I wait for someone to host this on https://cdn.coronastatus.eu before I update the src and SRI hash?

**Comment by adriaandotcom** [comment]
Added it to our CDN:
- https://cdn.coronastatus.eu/libraries/flatpickr-4.6.3.js
- https://cdn.coronastatus.eu/libraries/flatpickr-4.6.3.css

Please still use https://www.srihash.org and these attributes, just to be sure:

```html
<script
  src="https://cdn.coronastatus.eu/libraries/..."
  integrity="sha384-..."
  crossorigin="anonymous"
  referrerpolicy="no-referrer"
></script>
```

```html
<link
  rel="stylesheet"
  href="https://cdn.coronastatus.eu/libraries/..."
  integrity="sha384-..."
  crossorigin="anonymous"
  referrerpolicy="no-referrer"
>
```

After your PR I can do it for other external libs as well.

**Comment by jinjie** [comment]
> * Should we format the date that is displayed to the user after it is selected? Seems like Flatpickr has support for this. We could just add the format to `/locales/xx.json`.

We should. However, need to load the respective i10n .js files for the localisation. Not very sure how I can do this.

> * Could you try to fix the input field on iOS devices? Don't know if it looks the same on Androids.

I do not have an iOS mobile device. But it looks fine on Safari for me.

I am pretty new to building a site on node... Pardon me if my skills are limited at this time 😆. I can help within my limit, but I am contributing to help and learn more about node..

**Comment by fossecode** [comment]
> > * Should we format the date that is displayed to the user after it is selected? Seems like Flatpickr has support for this. We could just add the format to `/locales/xx.json`.
> 
> We should. However, need to load the respective i10n .js files for the localisation. Not very sure how I can do this.
> 
> > * Could you try to fix the input field on iOS devices? Don't know if it looks the same on Androids.
> 
> I do not have an iOS mobile device. But it looks fine on Safari for me.
> 
> I am pretty new to building a site on node... Pardon me if my skills are limited at this time 😆. I can help within my limit, but I am contributing to help and learn more about node..

Ok np, I can take a look at the iOS-date later :) You can render the date-format backend like this:
```html
<script src="https://cdn.jsdelivr.net/npm/flatpickr"></script>
<script src="/static/form.js?v=<%= locals.cacheKey %>"></script>
<script>
  flatpickr(document.querySelectorAll('[type=date]'), {
    altInput: true,
    altFormat: '<%= __(`F j, Y`) %>',
    dateFormat: 'Y-m-d'
  });
</script>
```
Then people can change the value of `'F j, Y'` as they see fit

**Review by fossecode — APPROVED** [review]
Awesome! @adriaanvanrossum what do you think about using `F j, Y` as default? https://flatpickr.js.org/examples/#human-friendly-dates

I think we should just merge this with a default  `altFormat` (as it is now), and then people can add custom formatting for their locales if they want that.

**Review by adriaandotcom — APPROVED** [review]

**Pull request merged**